edx/edx-enterprise
# Adding course completion flow to the Canvas integrated channel.

**Merge checklist:**
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

**Post merge:**
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

[Jira Ticket ENT 3085](https://openedx.atlassian.net/secure/RapidBoard.jspa?rapidView=630&projectKey=ENT&modal=detail&selectedIssue=ENT-3085)

**AC for ticket:**
- When an enterprise learner enrolls in and subsequently completes a course within the learner's enterprise catalog, the learners grade should be transmitted and available within the learner's Canvas course `grades` section.

- How this PR accomplishes the **AC**:
   - The client will export and transmit the learner's `percent grade` as a submitted grade to an automatically generated "Final Grade" assignment on Canvas.  
   - This will look something like: 
![image](https://user-images.githubusercontent.com/67655836/92268251-0a23e200-eeb0-11ea-933d-3e9f406508e6.png)

To test locally (gl hf):
- follow the wonderfully laid out steps to create a course -> [thank you jeff](https://openedx.atlassian.net/wiki/spaces/~479616267/pages/1789231216/Creating+a+Course+Locally+in+Publisher+and+adding+to+an+Enterprise+Catalog)
- Add your new course to the enterprise course catalog and add a course query to one of your enterprise customers
- Transmit the course metadata to Canvas so that it appears within courses under the admin Canvas user.
- Make sure you have enterprise learners associated with the enterprise customers
   - in the admin portal, click on your enterprise customer, then click `manage learners` in the top right, with the form, enroll your learner  
- Sign in as the learner to lms/dashboard and complete the course
- run the `transmit_learner_data` command in enterprise.
